tests/std/tests/LWG2510_tag_classes/test.compile.pass.cpp

```diff
@@ -0,0 +1,207 @@
+    STATIC_ASSERT(is_nothrow_constructible_v<T, T&>);
+    STATIC_ASSERT(is_nothrow_constructible_v<T, const T>);
+#if _HAS_CXX20
+    STATIC_ASSERT(is_nothrow_convertible_v<T&, T>);
+    STATIC_ASSERT(is_nothrow_convertible_v<const T&, T>);
+    STATIC_ASSERT(is_nothrow_convertible_v<T, T>);
+    STATIC_ASSERT(is_nothrow_convertible_v<const T, T>);
+#endif // _HAS_CXX20
+    STATIC_ASSERT(is_nothrow_copy_assignable_v<T>);
+    STATIC_ASSERT(is_nothrow_move_assignable_v<T>);
+    STATIC_ASSERT(is_nothrow_assignable_v<T&, T&>);
+    STATIC_ASSERT(is_nothrow_assignable_v<T&, const T>);
+
+    STATIC_ASSERT(is_trivially_default_constructible_v<T>);
+    STATIC_ASSERT(is_trivially_copy_constructible_v<T>);
+    STATIC_ASSERT(is_trivially_move_constructible_v<T>);
+    STATIC_ASSERT(is_trivially_constructible_v<T, T&>);
+    STATIC_ASSERT(is_trivially_constructible_v<T, const T>);
+    STATIC_ASSERT(is_trivially_copy_assignable_v<T>);
+    STATIC_ASSERT(is_trivially_move_assignable_v<T>);
+    STATIC_ASSERT(is_trivially_assignable_v<T&, T&>);
+    STATIC_ASSERT(is_trivially_assignable_v<T&, const T>);
+    STATIC_ASSERT(is_trivially_destructible_v<T>);
+    STATIC_ASSERT(is_trivially_copyable_v<T>);
+
+    STATIC_ASSERT(is_standard_layout_v<T>);
+    STATIC_ASSERT(is_empty_v<T>); // only guaranteed for allocator_arg_t and piecewise_construct_t
+
+    // ABI-specific, seemingly agreed among mainstream implementations
+    STATIC_ASSERT(sizeof(T) == 1);
+}
+
+void check_tag_classes() { // COMPILE-ONLY
+    check_tag_class<nothrow_t>();
+    check_tag_class<allocator_arg_t>();
+    check_tag_class<piecewise_construct_t>();
+#ifndef _M_CEE_PURE
+    check_tag_class<adopt_lock_t>();
+    check_tag_class<defer_lock_t>();
+    check_tag_class<try_to_lock_t>();
+#endif // _M_CEE_PURE
+
+#if _HAS_CXX17
+    check_tag_class<in_place_t>();
+    check_tag_class<in_place_index_t<0>>();
+    check_tag_class<in_place_index_t<42>>();
+    check_tag_class<in_place_index_t<static_cast<size_t>(-1)>>();
+    check_tag_class<in_place_type_t<int>>();
+    check_tag_class<in_place_type_t<in_place_type_t<int>>>();
+    check_tag_class<in_place_type_t<void>>();
+    check_tag_class<in_place_type_t<void()>>();
+
+    // LWG-4273 "Standard execution policy types should be conventional tag class types"
+    check_tag_class<execution::sequenced_policy>();
+    check_tag_class<execution::parallel_policy>();
+    check_tag_class<execution::parallel_unsequenced_policy>();
+#endif // _HAS_CXX17
+
+#if _HAS_CXX20
+    check_tag_class<destroying_delete_t>();
+    check_tag_class<nostopstate_t>();
+    check_tag_class<chrono::last_spec>();
+
+    // LWG-4273 "Standard execution policy types should be conventional tag class types"
+    check_tag_class<execution::unsequenced_policy>();
+#endif // _HAS_CXX20
+
+#if _HAS_CXX23
+    check_tag_class<from_range_t>();
+    check_tag_class<unexpect_t>();
+    // TODO: Cover sorted_{equivalent,unique}_t once flat_meow are implemented.

```

Comment:
`sorted_{equivalent,unique}_t` makes it harder to search for these identifiers and doesn't save a lot of space. Let's spell them out.